lib/action_controller/serialization.rb

```diff
@@ -23,7 +23,19 @@ def serialization_scope(scope)
     end
 
     def namespace_for_serializer
-      @namespace_for_serializer ||= self.class.parent unless self.class.parent == Object
+      @namespace_for_serializer ||= namespace_for_class(self.class) unless namespace_for_class(self.class) == Object
+    end
+
+    if Module.method_defined?(:module_parent)
+      def namespace_for_class(klass)
+        klass.module_parent
+      end
+    elsif Module.method_defined?(:parent)
+      def namespace_for_class(klass)
+        klass.parent
+      end
+    else
+      fail 'Module must define either parent or module_parent'
```

Comment:
I would prefer to have it this way:
```
def namespace_for_class(klass)
  if Module.method_defined?(:module_parent)
    klass.module_parent
  else
    klass.parent
  end
end
```
I dont think the `fail` case would ever happen as Rails is supposed to provide either `Module.parent` upto Rails 6.0 at least and `Module.module_parent` since Rails 6.1 afaik

Reply:
I agree - adjusting this now. Look for the commit shortly.